### DISCUSSION TITLE
Bun v0.0.55

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#79] [Announcements]
This release is mostly just bug fixes.

### CommonJS <> ESM interop reliability improvements

Bun now preserves [live bindings](https://stackoverflow.com/questions/52211309/what-does-it-mean-by-live-bindings) when referencing bundled code. This fixes a number of subtle bugs in different packages. 

This also updates the preferred extension order depending on how code is imported. Previously, `.mjs` files were ignored and that was silly. Now, when you use `import`, Bun will attempt to load `.mjs` or `.mts` before trying `.js` or `.cjs`

```zig
// CommonJS or `require`, `require.resolve`
// configurable via --extension-order CLI flag
pub const ExtensionOrder = [_]string{
    ".tsx",
    ".ts",
    ".jsx",
    ".cts",
    ".cjs",
    ".js",
    ".mjs",
    ".mts",
    ".json",
};

// ES Modules or `import`
pub const ModuleExtensionOrder = [_]string{
    ".tsx",
    ".jsx",
    ".mts",
    ".ts",
    ".mjs",
    ".js",
    ".cts",
    ".cjs",
    ".json",
};
```

### Template literal parsing bug

Before, this code caused an assertion failure in Bun's JavaScript parser due to function being defined in the template tag:

```js
import styled from 'styled-components'

export const HoverableBox = styled.div.attrs<{
  disabled?: boolean
}>(({ disabled }) => ({
  cursor: disabled ? undefined : 'pointer',
}))<{ disabled?: boolean }>`
  ${({ disabled }) => (disabled ? 'pointer-events: none;' : '')}
`
```

The problem was related to when scopes for template literal tags were visited. This has been fixed.

### `try` & `require()`

Previously, this code would produce a build error:

```js
try {
  require("this-package-should-not-exist");
} catch (exception) {}

try {
  await import("this-package-should-not-exist");
} catch (exception) {}

import("this-package-should-not-exist").then(
  () => {},
  () => {}
);
```

In each of these cases, errors are caught at runtime, so it should not produce a build error.

### Top-level await is now enabled

Top-level await will no longer error when targeting browsers, however you should know that since Bun does not transpile for backwards compatibility, you will need to be sure it is safe to use in the target environment.

Using `module.exports` with top-level await is undefined behavior

<hr /><em>This discussion was created from the release <a href='https://github.com/Jarred-Sumner/bun/releases/tag/bun-v0.0.55'>Bun v0.0.55</a>.</em>

**Comment by DrSensor** [comment]
**Note for Archlinux user**: somehow Archlinux still stuck on glibc 2.33 so you need to build glibc by yourself or compile bun on your machine.

**Reply by Jarred-Sumner** [reply]
can you give this version a try and let me know if it still reports an issue?
[bun-linux-x64.zip](https://github.com/Jarred-Sumner/bun/files/7745000/bun-linux-x64.zip)

**Reply by DrSensor** [reply]
It works! No issue so far.
